.github/workflows/release.yml

```diff
@@ -0,0 +1,69 @@
+name: Release
+on:
+  # push:
+  #   tags:
+  #     - v*
+  release:
+    types:
+      - published
+
+env:
+  ELIXIR_OPTS: "--warnings-as-errors"
+  ERLC_OPTS: "warnings_as_errors"
+  LANG: C.UTF-8
+
+jobs:
+  release_pre_built:
+    strategy:
+      fail-fast: true
+      matrix:
+        otp_release: ["24", "23", "22"]
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 50
+      - uses: erlef/setup-beam@v1
+        with:
+          otp-version: ${{ matrix.otp_release }}
+      - name: Build Elixir Release
+        run: |
+          vsn=$(cat VERSION)
+          make Precompiled.zip
+          mv Precompiled-v${vsn}.zip elixir-${vsn}-otp-${{ matrix.otp_release }}.zip
+          shasum -a 1 elixir-${vsn}-otp-${{ matrix.otp_release }}.zip > elixir-${vsn}-otp-${{ matrix.otp_release }}.zip.sha1
+          shasum -a 256 elixir-${vsn}-otp-${{ matrix.otp_release }}.zip > elixir-${vsn}-otp-${{ matrix.otp_release }}.zip.sha256
+      - name: Upload Pre-built
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          vsn=$(cat VERSION)
+          gh release upload --clobber \
+            "v${vsn}" elixir-${vsn}-otp-${{ matrix.otp_release }}.zip \
+            elixir-${vsn}-otp-${{ matrix.otp_release }}.zip.sha{1,256}
+
+  release_docs:
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 50
+      - uses: erlef/setup-beam@v1
+        with:
+          otp-version: 24
+      - name: Build Elixir
+        run: |
+          make compile
```

Comment:
it's a bit wasteful to compile Elixir _again_ just to build docs. What do you think about re-using the builds from the previous job? The matrix step could look like this:

```yaml
      matrix:
        include:
          - otp: 22
          - otp: 23
          - otp: 24
            build_docs: build_docs

      # ...

      - run: |
          # ...
        if: ${{ matrix.build_docs }}

```

Reply:
This look better than my current setup. I’ll try it tomorrow since this is near midnight in my timezone. 😴